### ISSUE TITLE
Installing Hybrid Equations Toolbox gives errors

### ISSUE THREAD

**Issue by timaiken** [pnanez/HyEQ_Toolbox#89]
This is wil installing Hybrid Equations Toolbox on Linux immediately after installing MatLab + Simulink. I don't know if this error is innocuous.

(base) ThinkPad-P52s:~/matlab$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.4 LTS
Release:	20.04
Codename:	focal

Below is the output from the MATLAB Command Window
```
-----------------------------------------------------------------------------
	Trial License -- for use to evaluate programs for possible purchase as an end-user only.

com.jogamp.opengl.GLException: X11GLXDrawableFactory - Could not initialize shared resources for X11GraphicsDevice[type .x11, connection :0, unitID 0, handle 0x0, owner false, ResourceToolkitLock[obj 0x43853aed, isOwner false, <260eae98, b9c115c>[count 0, qsz 0, owner <NULL>]]]
	at jogamp.opengl.x11.glx.X11GLXDrawableFactory$SharedResourceImplementation.createSharedResource(X11GLXDrawableFactory.java:326)
	at jogamp.opengl.SharedResourceRunner.run(SharedResourceRunner.java:297)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at jogamp.opengl.GLContextImpl.makeCurrent(GLContextImpl.java:688)
	at jogamp.opengl.GLContextImpl.makeCurrent(GLContextImpl.java:580)
	at jogamp.opengl.x11.glx.X11GLXDrawableFactory$SharedResourceImplementation.createSharedResource(X11GLXDrawableFactory.java:297)
	... 2 more
Configuring the HyEQ Toolbox. This may take a minute to start while Simulink loads...
Updated Simulink model "HyEQ_Library" to the current version of Simulink.
Updated Simulink model "hybrid.examples.analog_to_digital_converter.adc" to the current version of Simulink.
Updated Simulink model "hybrid.examples.behavior_in_C_intersection_D.hybrid_priority" to the current version of Simulink.
Updated Simulink model "hybrid.examples.bouncing_ball.bouncing_ball" to the current version of Simulink.
Updated Simulink model "hybrid.examples.bouncing_ball_with_adc.ball_with_adc" to the current version of Simulink.
Updated Simulink model "hybrid.examples.bouncing_ball_with_input.ball_with_input" to the current version of Simulink.
Updated Simulink model "hybrid.examples.bouncing_ball_with_input.ball_with_input2" to the current version of Simulink.
Updated Simulink model "hybrid.examples.coupled_subsystems.coupled" to the current version of Simulink.
Updated Simulink model "hybrid.examples.finite_state_machine.fsm" to the current version of Simulink.
Updated Simulink model "hybrid.examples.fireflies.fireflies" to the current version of Simulink.
Updated Simulink model "hybrid.examples.mobile_robot.mobile_robot" to the current version of Simulink.
Updated Simulink model "hybrid.examples.network_estimation.network_est" to the current version of Simulink.
Updated Simulink model "hybrid.examples.vehicle_on_constrained_path.vehicle_on_path" to the current version of Simulink.
Updated Simulink model "hybrid.examples.zero_order_hold.zoh" to the current version of Simulink.
Updated Simulink model "hybrid.examples.zoh_feedback_control.zoh_feedback" to the current version of Simulink.
Enabled autocomplete information for the Hybrid Equations Toolbox.
MATLAB has experienced a low-level graphics error, and may not have drawn correctly.
Read about what you can do to prevent this issue at Resolving Low-Level Graphics Issues then restart MATLAB.
To share details of this issue with MathWorks technical support,
please include this file with your service request.
Tests skipped.

The Hybrid Equations Toolbox is ready to use.
```

**Comment by pwintz** [comment]
What version of MATLAB/Simulink are you using? Can you open and run Simulink models that don't use the HyEQ Toolbox?

**Comment by pwintz** [comment]
In the past, I've had trouble getting older versions of Simulink to work on Linux. Please check that your setup satisfies the [system requirements](https://www.mathworks.com/support/requirements/matlab-linux.html).

**Comment by timaiken** [comment]
Hi Paul,
Here is some more information.
When MATLAB starts up I get a couple of error messages:
(base) ***@***.***:~/matlab/R2022b$ matlabGtk-Message: 16:38:47.648: Failed to load module "overlay-scrollbar"Gtk-Message: 16:38:47.653: Failed to load module "canberra-gtk-module"
The (functional) issue is that I don't see plots correctly. Most are blank. Changing LINEWIDTH from 1 results in showing only part of the line. The problem is most likely related to graphics driver or library (openGL?).  In the command window see the error:

MATLAB has experienced a low-level graphics error, and may not have drawn correctly.Read about what you can do to prevent this issue at Resolving Low-Level Graphics Issues then restart MATLAB.To share details of this issue with MathWorks technical support,please include this file with your service request.

Looking at the system requirements link, it seems that everything is pretty much satisfied:
Linux Distribution
(base) ***@***.***:~/matlab$ lsb_release -aNo LSB modules are available.Distributor ID: UbuntuDescription: Ubuntu 20.04.4 LTSRelease: 20.04Codename: focal

Processor
(base) ***@***.***:~/matlab$ lscpuArchitecture:                    x86_64CPU op-mode(s):                  32-bit, 64-bitByte Order:                      Little EndianAddress sizes:                   39 bits physical, 48 bits virtualCPU(s):                          8On-line CPU(s) list:             0-7Thread(s) per core:              2Core(s) per socket:              4Socket(s):                       1NUMA node(s):                    1Vendor ID:                       GenuineIntelCPU family:                      6Model:                           142Model name:                      Intel(R) Core(TM) i7-8650U CPU @ 1.90GHzStepping:                        10CPU MHz:                         3691.140CPU max MHz:                     4200.0000CPU min MHz:                     400.0000BogoMIPS:                        4199.88Virtualization:                  VT-xL1d cache:                       128 KiBL1i cache:                       128 KiBL2 cache:                        1 MiBL3 cache:                        8 MiBNUMA node0 CPU(s):               0-7...

RAM
(base) ***@***.***:~/matlab$ grep MemTotal /proc/meminfo MemTotal:       32589116 kB

Disk Space
(base) ***@***.***:~/matlab$ dfFilesystem     1K-blocks     Used Available Use% Mounted onudev            16264988        0  16264988   0% /devtmpfs            3258912     2448   3256464   1% /run/dev/nvme0n1p2 490556632 74248708 391362416  16% /tmpfs           16294556   611444  15683112   4% /dev/shmtmpfs               5120        4      5116   1% /run/locktmpfs           16294556        0  16294556   0% /sys/fs/cgroup

Graphics Card
(base) ***@***.***:~/matlab$ lspci | grep VGA00:02.0 VGA compatible controller: Intel Corporation UHD Graphics 620 (rev 07)

Graphics Library/Driver
MATLAB rendererinfo returnsans = struct with fields:
    GraphicsRenderer: 'OpenGL Hardware'
              Vendor: ''
             Version: ''
      RendererDevice: 'None'
             Details: [1×1 struct]

OpenGL version
(base) ***@***.***:~/matlab$ glxinfo | grep "OpenGL version"OpenGL version string: 4.6 (Compatibility Profile) Mesa 21.2.6


More Info
As mentioned previously, I receive the message below.
>> opengl hardwarebasicMATLAB has experienced a low-level graphics error, and may not have drawn correctly.Read about what you can do to prevent this issue at Resolving Low-Level Graphics Issues then restart MATLAB.To share details of this issue with MathWorks technical support,please include this file with your service request.
The file of Java stack traces referenced above is attached.
One more thing.  Running matlab -softwareopengl seems to work OK, at least so far.  I guess this is my workaround for now. But I am just getting started. We'll see how things go.
Thanks again for your attention.
Cheers,
Tim
-----------------------------------------------------------------------------
    On Friday, January 20, 2023 at 02:07:40 AM PST, Paul Wintz ***@***.***> wrote:  
 
 


In the past, I've had trouble getting older versions of Simulink to work on Linux. Please check that your setup satisfies the system requirements.

**Comment by pwintz** [comment]
Ok. This appears to not be a problem with the HyEQ Toolbox, and you found a workaround, so I'm going to close this issue. You might want to run `hybrid.tests.run('essential')` to check that everything is working OK. Let me know if you run into any more problems.
